packages/calcite-components/src/utils/locale.ts

```diff
@@ -409,8 +409,8 @@ export class NumberStringFormat {
     } as Intl.NumberFormatOptions).formatToParts(-12345678.9);
 
     this._actualGroup = parts.find((d) => d.type === "group").value;
-    // change whitespace group characters that don't render correctly
-    this._group = this._actualGroup.trim().length === 0 ? " " : this._actualGroup;
+    // change whitespace group characters that don't render correctly to a unicode no-break space
+    this._group = this._actualGroup.trim().length === 0 || this._actualGroup == "\u0020" ? "\u00A0" : this._actualGroup;
```

Comment:
It might be prettier to use html entities here instead of unicode. What do you think?

```
this._actualGroup === " " ? "&nbsp;" : this._actualGroup;
```

Reply:
Yeah I agree that's nicer for the nbsp, but I like being very explicit about what is in `" "`, since there are a ton of white space characters that could have been copy/pasted in there. And then I didn't want to mix and match unicode and HTML entities. But I don't feel strongly about it if you'd like me to switch. Also story incoming